src/components/grouped-character/grounded-character.test.tsx

```diff
@@ -0,0 +1,604 @@
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={onBlur}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "12-34-");
+  await input.setSelectionRange(6, 6);
+  await user.keyboard("{backspace}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(4);
+  expect(input.selectionEnd).toBe(4);
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "12-");
+  await input.setSelectionRange(3, 3);
+  await user.keyboard("{backspace}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(1);
+  expect(input.selectionEnd).toBe(1);
+});
+
+test("pressing a character at the point where a separator should appear moves cursor forward if the cursor positioned in the middle of input value", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "1222");
+  await input.setSelectionRange(6, 6);
+  await user.tab();
+
+  expect(input.selectionStart).toBe(7);
+  expect(input.selectionEnd).toBe(7);
+});
+
+test("pressing backspace after a separator moves the cursor backwards to positions", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("1231231");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
```

Comment:
**comment/praise**: I think these arrowleft/arrowright combinations are a better way of getting the cursor to a specific position than using `setSelectionRange`: it's a much closer fit to what a user might do.